llvm/include/llvm/IR/IntrinsicInst.h

```diff
@@ -341,6 +341,13 @@ class DbgVariableIntrinsic : public DbgInfoIntrinsic {
     return getIntrinsicID() == Intrinsic::dbg_declare;
   }
 
+  /// Does this describe the value of a local variable. True for dbg.value,
+  /// but not dbg.declare, which describes its address, or dbg.assign, which
+  /// describes a combination of the variable's value and address.
+  bool isValueOfVariable() const {
+    return getIntrinsicID() == Intrinsic::dbg_value;
```

Comment:
how is this condition true for a dbg.assign (as indicated in the comment), does it share the same intrinsic ID?

Reply:
I read this as being that it describes dbg.value, but not (dbg.declare or dbg.assign).

Reply:
@SLTozer is right, maybe I need to word it better